### PR TITLE
Solving com_search Notice

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue #13960

### Summary of Changes
Check if a menu exists.
### Testing Instructions
See #13960

After patch the PHP logs will no more display the Notice.

### CODE REVIEW

File: components/com_search/controller.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ public function search()
 		$item = $menu->getItem($post['Itemid']);
 
 		// The requested Item is not a search page so we need to find one
-		if ($item->component != 'com_search' || $item->query['view'] != 'search')
+		if (!empty($item) && ($item->component != 'com_search' || $item->query['view'] != 'search'))
```
Comment (joomdonation) [inline components/com_search/controller.php]: I think change this line of code to

```php
 if ($item && ($item->component != 'com_search' || $item->query['view'] != 'search'))
```
will make it easier to read

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on bc05de9503f65d3fcafd0f04ef6e909988ecbc30<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13961">issues.joomla.org/tracker/joomla-cms/13961</a>.</sub>

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on bc05de9503f65d3fcafd0f04ef6e909988ecbc30<br /><br />Real test success but it should be nicer to not use empty function as mentioned by @joomdonation.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13961">issues.joomla.org/tracker/joomla-cms/13961</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13961">issues.joomla.org/tracker/joomla-cms/13961</a>.</sub>

**Comment by joomdonation** [comment]
@jeckodevelopment @rdeutz It would be nice if the change suggested by me (agreed by @csthomas) implemented before this PR is set to RTC and merged.

**Comment by infograf768** [comment]
sorry folks, read this too late.
welcome to a new PR simplifying this one as proposed.

**Comment by joomdonation** [comment]
@infograf768 Could you please make a new PR for this small change?

**Comment by infograf768** [comment]
Please test https://github.com/joomla/joomla-cms/pull/14035

**Pull request merged**